Named version for profile commands, and better docs (#470)
aws/eks-anywhere-packages: generatebundlefile/README.md

This will output a sample bundle file to ./output.

### Public Promotion to another Account

```sh

> generatebundlefile --private-profile "profile-name" --input data/sample_input.yaml

Reviewer: This command, and the one below it are exactly the same. I'm guessing that's a typo? 

If not, it might be worth calling out, and explaining how the same command can provide different results.

Author: good callout, copy/paste error.

```

> This command will move **Only** the helm chart from the listed input files to the target **public** ECR in another account.

Reviewer: ```suggestion
The command above will move **Only** the helm chart from the listed input files to the target **public** ECR in another account.
```

Best to be more explicit, since there are commands both above and below this statement, and it's not immediately obvious to which command it is referring.

### Private Promotion to another Account

```sh
generatebundlefile --private-profile "profile-name" --input data/sample_input.yaml
```

This command will move **Only** the images from the listed helm charts in the input files to the target **private** ECR in another account.

### Package Promotion

To promote a package from a private ECR to public you need the repository name. This repository must contain a Helm chart built by the process in the eks-anywhere-build-tooling git repository.

> This will **both** for the helm chart, and the imgaes required to the public ECR of the calling user.

Reviewer: This sentence is very confusing. Could you try re-wording it? I'd take a guess at re-wording it, but I'm really not sure what you meant.

> **This command only supports promotion of the latest helm chart from the private repository.**

Reviewer: ```suggestion
**Note: The command below only supports promotion of the latest helm chart from the private repository.**
```

```sh
generatebundlefile --promote hello-eks-anywhere
```

### Input File Supported Formats

Currently the following formats can be used as input files each of the following command flags.

--key alias/signingPackagesKey
--private-profile
--public--profile

#### Private Registry

```yaml
name: "v1-21-1001"
kubernetesVersion: "1.21"
packages:
  - org: aws-containers
    projects:
      - name: hello-eks-anywhere
        repository: hello-eks-anywhere
        registry: 646717423341.dkr.ecr.us-west-2.amazonaws.com
        versions:
            - name: latest
```

#### Public Registry

```yaml
name: "v1-21-1001"
kubernetesVersion: "1.21"
packages:
  - org: aws-containers
    projects:
      - name: hello-eks-anywhere
        repository: hello-eks-anywhere
        registry: public.ecr.aws/eks-anywhere
        versions:
            - name: latest
```

#### Latest Version

> The "latest" tag will tell the program to use a timestamp lookup, and find the most recently pushed helm chart for information, even if it does have the actual latest tag.

Reviewer: ```suggestion
The "latest" tag will tell the program to use a timestamp lookup, and find the most recently pushed helm chart for information, even if that helm chart doesn't have the latest tag.
```
